### PR TITLE
Unpin EF dependencies then grab the latest

### PR DESCRIPTION
**Pull request by dougbu**
- partially reverts 58c67727dfb3
- temporarily moves CoreFx, core-setup and Extensions dependencies
  - will update EF Core repo before finalizing this PR

### CODE REVIEW

**Comment by dougbu** [comment]
@ajcvickers lots of MusicStore failures in https://dev.azure.com/dnceng/public/_build/results?buildId=206727 Any suggestions?

**Comment by ajcvickers** [comment]
Ah, man. I didn't realize we were still building MusicStore. I will take a look.

**Comment by dougbu** [comment]
@Eilon / @Pilchie / @aspnet/build I'm not sure `area-infrastructure` makes sense here because the PR's about including EF in this release and testing against what we're shipping. `area-platform` might be more correct.

But, do we need an `area-ef` or perhaps `area-musicstore` label in this repo?

**Comment by Eilon** [comment]
From my perspective:
- The area labels are to help make sure the right set of people look at an issue/PR
- If area labels are wrong, anyone is free to update them. Updating them will help the Machine Learning system do better in the future.

**Comment by ajcvickers** [comment]
@dougbu Pushed MusicStore fixes

**Review by ajcvickers — APPROVED** [review]

**Comment by ajcvickers** [comment]
@dougbu Note that @smitpatel [merged a race condition fix](https://github.com/aspnet/EntityFrameworkCore/pull/15880) that we may need to take here to get the tests passing reliably.

**Comment by dougbu** [comment]
Failures are now focused on 2 test templates:
- `Templates.Test.SpaTemplateTest.ReactTemplateTest.ReactTemplate_IndividualAuth_NetCore_LocalDb`
- `Templates.Test.SpaTemplateTest.AngularTemplateTest.AngularTemplate_IndividualAuth_Works_LocalDb`

I suspect the template itself needs to be updated to match the latest from EF. @ajcvickers could you please have a look? @ryanbrandenburg is the templates guy if you're not sure about the context.

----
Separately, the following tests failed only on macOS. Is there a race in these tests?
- `MusicStore.Controllers.CheckoutControllerTest.AddressAndPayment_RedirectToCompleteWhenSuccessful`
- `MusicStore.Components.CartSummaryComponentTest.CartSummaryComponent_Returns_CartedItems`
  - Guess it's also possible Sqlite does something odd on macOS, leading at least to this failure

**Comment by ajcvickers** [comment]
@dougbu Will take a look

**Comment by dougbu** [comment]
:up::date: to rebase on 'master' and see if @smitpatel's fix cleans up the tests

**Comment by ajcvickers** [comment]
@dougbu It looks like all the changes I pushed here have been removed!

**Comment by dougbu** [comment]
@ajcvickers yeah, that was my fault. Fortunately I still had 6cf65d786b67c22288024a1e48b6d736d305846b on my machine (GitHub lost that commit) and have restored it.

**Comment by dougbu** [comment]
With the addition of a spurious Libuv test failure, looks like the two tests that failed consistently are still failing:
- `Templates.Test.SpaTemplateTest.ReactTemplateTest.ReactTemplate_IndividualAuth_NetCore_LocalDb`
- `Templates.Test.SpaTemplateTest.AngularTemplateTest.AngularTemplate_IndividualAuth_Works_LocalDb`

**Comment by ajcvickers** [comment]
@dougbu @ryanbrandenburg I can't get the ProjectTemplates tests to run locally, even though everything else in AspNetCore is working. I keep seeing the following, even on master:
```
CSC : warning CS2008: No source files specified. [C:\aspnet\AspNetCore\src\SiteExtensions\LoggingBranch\LoggingBranch.csproj]
C:\aspnet\AspNetCore\.dotnet\x64\sdk\3.0.100-preview5-011568\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.ComposeStore.targets(347,5): warning MSB3026: Could not copy "C:\Users\Arthur\.nuget\packages\runtime.win-x64.microsoft.netcore.app\3.0.0-preview6-27730-01\runtimes\win-x64\lib\netcoreapp3.0\System.Collections.Immutable.dll" to "C:\aspnet\AspNetCore\artifacts\obj\LoggingBranch\Debug\net461\se\rs\win-x64\x64\netcoreapp3.0\System.Collections.Immutable.dll". Beginning retry 1 in 1000ms. The process cannot access the file 'C:\aspnet\AspNetCore\artifacts\obj\LoggingBranch\Debug\net461\se\rs\win-x64\x64\netcoreapp3.0\System.Collections.Immutable.dll' because it is being used by another process.  [C:\aspnet\AspNetCore\artifacts\obj\LoggingBranch\Debug\net461\se\rs.csproj]C:\aspnet\AspNetCore\.dotnet\x64\sdk\3.0.100-preview5-011568\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.ComposeStore.targets(347,5): warning MSB3026: Could not copy "C:\Users\Arthur\.nuget\packages\runtime.win-x64.microsoft.netcore.app\3.0.0-preview6-27730-01\runtimes\win-x64\lib\netcoreapp3.0\System.Net.Http.dll" to "C:\aspnet\AspNetCore\artifacts\obj\LoggingBranch\Debug\net461\se\rs\win-x64\x64\netcoreapp3.0\System.Net.Http.dll". Beginning retry 1 in 1000ms. The process cannot access the file 'C:\aspnet\AspNetCore\artifacts\obj\LoggingBranch\Debug\net461\se\rs\win-x64\x64\netcoreapp3.0\System.Net.Http.dll' because it is being used by another process.  [C:\aspnet\AspNetCore\artifacts\obj\LoggingBranch\Debug\net461\se\rs.csproj]C:\aspnet\AspNetCore\.dotnet\x64\sdk\3.0.100-preview5-011568\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.ComposeStore.targets(347,5): warning MSB3026: Could not copy "C:\Users\Arthur\.nuget\packages\runtime.win-x64.microsoft.netcore.app\3.0.0-preview6-27730-01\runtimes\win-x64\lib\netcoreapp3.0\System.Linq.Expressions.dll" to "C:\aspnet\AspNetCore\artifacts\obj\LoggingBranch\Debug\net461\se\rs\win-x64\x64\netcoreapp3.0\System.Linq.Expressions.dll". Beginning retry 1 in 100ms. The process cannot access the file 'C:\aspnet\AspNetCore\artifacts\obj\LoggingBranch\Debug\net461\se\rs\win-x64\x64\netcoreapp3.0\System.Linq.Expressions.dll' because it is being used by another process. The file is locked by: "MSBuild.exe (4244), Windows Defender Antivirus Service (5304)" [C:\aspnet\AspNetCore\artifacts\obj\LoggingBranch\Debug\net461\se\rs.csproj]
C:\aspnet\AspNetCore\.dotnet\x64\sdk\3.0.100-preview5-011568\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.ComposeStore.targets(347,5): warning MSB3026: Could not copy "C:\Users\Arthur\.nuget\packages\runtime.win-x64.microsoft.netcore.app\3.0.0-preview6-27730-01\runtimes\win-x64\lib\netcoreapp3.0\System.Linq.Expressions.dll" to "C:\aspnet\AspNetCore\artifacts\obj\LoggingBranch\Debug\net461\se\rs\win-x64\x64\netcoreapp3.0\System.Linq.Expressions.dll". Beginning retry 1 in 1000ms. The process cannot access the file 'C:\aspnet\AspNetCore\artifacts\obj\LoggingBranch\Debug\net461\se\rs\win-x64\x64\netcoreapp3.0\System.Linq.Expressions.dll' because it is being used by another process. The file is locked by: "MSBuild.exe (4244), MSBuild.exe (15428), Windows Defender Antivirus Service (5304)" [C:\aspnet\AspNetCore\artifacts\obj\LoggingBranch\Debug\net461\se\rs.csproj]C:\aspnet\AspNetCore\.dotnet\x64\sdk\3.0.100-preview5-011568\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.ComposeStore.targets(347,5): warning MSB3026: Could not copy "C:\Users\Arthur\.nuget\packages\runtime.win-x64.microsoft.netcore.app\3.0.0-preview6-27730-01\runtimes\win-x64\lib\netcoreapp3.0\System.Linq.Expressions.dll" to "C:\aspnet\AspNetCore\artifacts\obj\LoggingBranch\Debug\net461\se\rs\win-x64\x64\netcoreapp3.0\System.Linq.Expressions.dll". Beginning retry 1 in 1000ms. The process cannot access the file 'C:\aspnet\AspNetCore\artifacts\obj\LoggingBranch\Debug\net461\se\rs\win-x64\x64\netcoreapp3.0\System.Linq.Expressions.dll' because it is being used by another process. The file is locked by: "MSBuild.exe (15428), Windows Defender Antivirus Service (5304)" [C:\aspnet\AspNetCore\artifacts\obj\LoggingBranch\Debug\net461\se\rs.csproj]C:\aspnet\AspNetCore\artifacts\obj\LoggingBranch\Debug\net461\se\rs.csproj : error NU1603: Microsoft.AspNetCore.AzureAppServicesIntegration 3.0.0-dev depends on Microsoft.AspNetCore.Hosting (>= 3.0.0-dev) but Microsoft.AspNetCore.Hosting3.0.0-dev was not found. An approximate best match of Microsoft.AspNetCore.Hosting 3.0.0-preview-18571-0006 was resolved.
C:\aspnet\AspNetCore\artifacts\obj\LoggingBranch\Debug\net461\se\rs.csproj : error NU1603: Microsoft.AspNetCore.AzureAppServicesIntegration 3.0.0-dev depends on Microsoft.AspNetCore.Hosting (>= 3.0.0-dev) but Microsoft.AspNetCore.Hosting3.0.0-dev was not found. An approximate best match of Microsoft.AspNetCore.Hosting 3.0.0-preview-18571-0006 was resolved.
C:\aspnet\AspNetCore\artifacts\obj\LoggingBranch\Debug\net461\se\rs.csproj : error NU1603: Microsoft.AspNetCore.AzureAppServicesIntegration 3.0.0-dev depends on Microsoft.AspNetCore.Hosting (>= 3.0.0-dev) but Microsoft.AspNetCore.Hosting3.0.0-dev was not found. An approximate best match of Microsoft.AspNetCore.Hosting 3.0.0-preview-18571-0006 was resolved.
C:\aspnet\AspNetCore\artifacts\obj\LoggingBranch\Debug\net461\se\rs.csproj : error NU1603: Microsoft.AspNetCore.AzureAppServicesIntegration 3.0.0-dev depends on Microsoft.AspNetCore.Hosting (>= 3.0.0-dev) but Microsoft.AspNetCore.Hosting3.0.0-dev was not found. An approximate best match of Microsoft.AspNetCore.Hosting 3.0.0-preview-18571-0006 was resolved.
C:\aspnet\AspNetCore\artifacts\obj\LoggingBranch\Debug\net461\se\rs.csproj : error NU1603: Microsoft.AspNetCore.AzureAppServicesIntegration 3.0.0-dev depends on Microsoft.AspNetCore.Hosting (>= 3.0.0-dev) but Microsoft.AspNetCore.Hosting3.0.0-dev was not found. An approximate best match of Microsoft.AspNetCore.Hosting 3.0.0-preview-18571-0006 was resolved.
C:\aspnet\AspNetCore\artifacts\obj\LoggingBranch\Debug\net461\se\rs.csproj : error NU1603: Microsoft.AspNetCore.AzureAppServicesIntegration 3.0.0-dev depends on Microsoft.AspNetCore.Hosting (>= 3.0.0-dev) but Microsoft.AspNetCore.Hosting3.0.0-dev was not found. An approximate best match of Microsoft.AspNetCore.Hosting 3.0.0-preview-18571-0006 was resolved.
C:\aspnet\AspNetCore\artifacts\obj\LoggingBranch\Debug\net461\se\rs.csproj : error NU1603: Microsoft.AspNetCore.AzureAppServicesIntegration 3.0.0-dev depends on Microsoft.AspNetCore.Hosting (>= 3.0.0-dev) but Microsoft.AspNetCore.Hosting3.0.0-dev was not found. An approximate best match of Microsoft.AspNetCore.Hosting 3.0.0-preview-18571-0006 was resolved.
C:\aspnet\AspNetCore\artifacts\obj\LoggingBranch\Debug\net461\se\rs.csproj : error NU1603: Microsoft.AspNetCore.AzureAppServicesIntegration 3.0.0-dev depends on Microsoft.AspNetCore.Hosting (>= 3.0.0-dev) but Microsoft.AspNetCore.Hosting3.0.0-dev was not found. An approximate best match of Microsoft.AspNetCore.Hosting 3.0.0-preview-18571-0006 was resolved.
C:\aspnet\AspNetCore\artifacts\obj\LoggingBranch\Debug\net461\se\Microsoft.AspNetCore.AzureAppServices.HostingStartup\HostingStartup.csproj : error NU1603: Microsoft.AspNetCore.AzureAppServicesIntegration 3.0.0-dev depends on Microsoft.AspNetCore.Hosting (>= 3.0.0-dev) but Microsoft.AspNetCore.Hosting 3.0.0-dev was not found. An approximate best match of
Microsoft.AspNetCore.Hosting 3.0.0-preview-18571-0006 was resolved.
    6 Warning(s)
    9 Error(s)

Time Elapsed 00:01:13.34
msbuild.exe failed with exit code: 1
C:\aspnet\AspNetCore\src\ProjectTemplates>
```

**Comment by dougbu** [comment]
@ajcvickers is the information in the folder's [README.md](https://github.com/aspnet/AspNetCore/blob/master/src/ProjectTemplates/README.md) file helpful?

**Comment by dougbu** [comment]
In particular, run `build.cmd -all -pack` in the repository root if you haven't already

**Comment by ajcvickers** [comment]
@dougbu Yes, read that and tried that already. Running `build -all -pack` from the top level results in those errors. Running just `build -pack` from the top level works fine.

**Comment by dougbu** [comment]
Can you run the template tests after the successful `build -pack`?

The other possibility would be downloading the ProjectTemplates.E2ETests&hellip;.binlog file from the [Windows_Test_Templates_Logs artifacts](https://dev.azure.com/dnceng/_apis/resources/Containers/1669584?itemPath=Windows_Test_Templates_Logs&$format=zip) of https://dev.azure.com/dnceng/public/_build/results?buildId=208469 (The cross-plat template tests don't seem to produce bin logs for this test group.)

@ryanbrandenburg any ideas?

**Comment by ajcvickers** [comment]
@dougbu I found a way to reproduce the failures locally and I am investigating.

@ryanbrandenburg @dougbu  Summary of trying to get the tests to run locally: I found I could get tests to run with `build -pack -test` at the top level. Anything using `-all` fails. But doing this at the top-level is very slow, so I looked at the difference when trying to run the tests from `src\ProjectTemplates` and found if I remove `/t:BuildSharedFx` from `build.cmd` at this level, then the build passes, and at least the tests I care about are now running.

**Comment by dougbu** [comment]
@ajcvickers I picked up the latest from 'master' as well as newer EF and AspNetCore packages in my force-push to this branch. Probably won't fix the template tests but avoids conflicts with #10736.

**Comment by dougbu** [comment]
> when trying to run the tests from `src\ProjectTemplates` and found if I remove `/t:BuildSharedFx` from `build.cmd` at this level, then the build passes

That makes sense because without a new Shared Fx on your machine, the tests run using the old EF packages.

@aspnet/build and @ryanbrandenburg, once the Shared Fx is built once, is it possible to point the template tests to what was built previously?

**Comment by dougbu** [comment]
:up::date: to pull in the latest

@rynowak @pranavkm @ajaybhargavb should I worry about breaking changes from Blazor or AspNetCore-Tooling ❔

File: eng/Versions.props [hunk 1]
```diff
@@ -49,7 +49,7 @@
     <!-- Only listed explicitly to workaround https://github.com/dotnet/cli/issues/10528 -->
     <MicrosoftNETCorePlatformsPackageVersion>3.0.0-preview6.19279.8</MicrosoftNETCorePlatformsPackageVersion>
     <!-- Packages from aspnet/Blazor -->
-    <MicrosoftAspNetCoreBlazorMonoPackageVersion>0.10.0-preview6.19273.9</MicrosoftAspNetCoreBlazorMonoPackageVersion>
+    <MicrosoftAspNetCoreBlazorMonoPackageVersion>0.10.0-preview7.19303.2</MicrosoftAspNetCoreBlazorMonoPackageVersion>
```
Comment (smitpatel) [inline eng/Versions.props]: Is preview7 is correct here?

Comment (dougbu) [reply eng/Versions.props]: @natemcmaster ?

**Comment by ryanbrandenburg** [comment]
@dougbu, by point it at what was built previously, you mean within a single machine build the SharedFx, then build/test templates without additional fiddling? If so yes, that should work.

**Comment by dougbu** [comment]
Getting this in because 2nd build attempt was started accidentally and first one showed only a Yarn registry failure and a known-flaky test

**Comment by ajcvickers** [comment]
Thanks for all your help on this @dougbu. Sorry it has been so painful.

**Comment by dougbu** [comment]
Well, we're not done yet 😈. I need to recreate this PR for preview 6 after Tooling and EF Core official builds get new bits into the right `darc` channel.

**Comment by ajcvickers** [comment]
Okay, 🤞 then.

**Pull request merged**